### PR TITLE
NEW: Add Simple Zoomify Viewer For WSIReader

### PR DESCRIPTION
**Pull request by John-P**
Adds a  simple Zoomify viewer class.

Key Features
========

- Take a dictionary of `WSIReader` objects
- Display then as layers in the web browser (`localhost:5000`) by serving Zoomify tiles
- Togglable layers
- Adjustable opacity
- Layer re-ordering
- Mini-map
- Fullscreen
- Rotation (shift + alt) with reset button
- Mouse coordinates in baseline pixels (bottom right)
- Scale bar in microns
- Works with JP2 images but can be very slow and a little glitchy sometimes.

To Do
====

- [x] Rename class / module?
- [x] Unit tests

Dependencies
=========

- OpenLayers (BSD) and ol-ext (BSD-like, French version) Javascript for the frontend.
- Flask (BSD) for the server.

Demo
====

https://user-images.githubusercontent.com/4615004/144947393-aa492644-6012-48d9-b73b-5109e67393bd.mp4

Update: Added Graticule & Screen Space Graticule (Grids)
------------------------------------------------------------------------------

https://user-images.githubusercontent.com/4615004/145240617-57f21e9a-faa8-4f9e-896b-b89c41c0df0b.mp4

Demo Code
-------------

```python
from tiatoolbox.wsiscore.wsireader import WSIReader
from tiatoolbox.visualization.tileserver import TileServer

wsi = WSIReader.open("/home/john/Downloads/CMU-1.svs")
wsi2 = WSIReader.open("/home/john/Downloads/JP2K-33003-1.svs")
app = TileServer(
    title="Testing TileServer",
    layers={
        "JPEG SVS": wsi,
        "JP2 SVS": wsi2,
    },
)
app.run()
```

 # Future Extensions

- Allow annotation store as layer to display vector polygons / rendered as tiles
- IPython support for embedding in notebooks
- Make the layers UI a bit prettier

### CODE REVIEW

**Comment by John-P** [comment]
As a result this has made some bugs in `VirtualWSIReader` in combination with `ZoomifyGenerator` apparent and will be useful for testing and debugging.

**Comment by John-P** [comment]
@shaneahmed Only a quick draft and simple tool but may be useful to motivate Zoomify tile generation and also as a way of debugging it / showing results in general.

 # Future Extensions

- [ ] Allow annotation store as layer to display vector polygons / rendered as tiles

**Review by shaneahmed — COMMENTED** [review]
Thanks @John-P  This will strengthen visualisation part. Can we add tests to fix coverage?

File: tiatoolbox/visualization/tileserver.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import numpy as np
+from flask import Flask, Response, send_file
+from flask.templating import render_template
+
+from tiatoolbox import data
+from tiatoolbox.tools.pyramid import ZoomifyGenerator
+from tiatoolbox.wsicore.wsireader import WSIReader
+
+
+class TileServer(Flask):
```
Comment (shaneahmed) [inline tiatoolbox/visualization/tileserver.py:35]: Can we add an example to visualise the images with the help of this server?

**Review by ghadjigeorghiou — APPROVED** [review]

**Review by shaneahmed — APPROVED** [review]

**Pull request merged**